### PR TITLE
BytesIO needes bytes not str

### PR DESCRIPTION
**Pull request by hysh**
Some commands were raising error due to BytesIO expecting bytes instead of str.

>   ./put.py --retention --gov=1d a
> Using 'cloudian' bucket 'r1bn' and user '0'
> 
> Putting Object retention for:  a None
> Traceback (most recent call last):
>   File "./put.py", line 428, in <module>
>     bypass_governance_retention=bypass_governance_retention)
>   File "/root/work/boto/boto/s3/key.py", line 2649, in set_retention
>     md5 = compute_md5(BytesIO(data))
> TypeError: a bytes-like object is required, not 'str'

### CODE REVIEW

File: boto/s3/key.py [hunk 1]
```diff
@@ -2530,7 +2530,7 @@ def set_remote_metadata(self, metadata_plus, metadata_minus, preserve_acl,
                             headers=headers)
 
     def select_object_content(self, data, headers=None):
-        md5 = compute_md5(BytesIO(data))
+        md5 = compute_md5(BytesIO(data.encode('utf-8')))
```
Comment (mozawa) [inline boto/s3/key.py]: Do we need this change?  I see this one in s3select.py.

`response = key.select_object_content(data.encode('utf-8'), headers=headers)`

**Comment by mozawa** [comment]
How about  `md5 = compute_md5(BytesIO(data))` in set_retention()?

**Comment by hysh** [comment]
> How about `md5 = compute_md5(BytesIO(data))` in set_retention()?

Misunderstood the if clause above it. Added the encode for set_retention() also. Thanks.

File: boto/s3/key.py (hunk 1 above)
Comment (hysh) [reply boto/s3/key.py]: OK not needed!

**Review by mozawa — APPROVED** [review]
LGTM

**Pull request merged**